docs/fundamentals/health-check.md

```diff
@@ -0,0 +1,238 @@
+# Health and readiness check of Jina Services
+
+(health-check-executor)=
+## Health check of an Executor
+
+Executors run as microservices exposing `grpc` endpoints. To give information to the outside world about their health and their readiness to receive requests,
+Executors expose a [grpc health check](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) service which can be used by different orchestrators like docker-compose, 
+or by other services like load-balancers. Jina itself when run locally uses this service to make sure that each Executor is ready to receive traffic.
+
+To see how to use it, let's start a Flow inside a Terminal and let it blocked to serve requests:
+
+```python
+from jina import Flow
+
+f = Flow(protocol='grpc', port=12345).add(port=12346)
+with f:
+    f.block()
+```
+
+On another terminal, we can install or download [grpcurl](https://github.com/fullstorydev/grpcurl) image to be able to send `rpc` requests to our services.
+
+```bash
+docker pull fullstorydev/grpcurl:latest
+docker run --network='host' fullstorydev/grpcurl -plaintext 127.0.0.1:12346 grpc.health.v1.Health/Check
+```
+
+```text
+{
+  "status": "SERVING"
+}
+```
+
+(health-check-gateway)=
+## Health check of the Gateway
+
+The same way individual Executors expose endpoints for orchestrators, clients or other services to check their availability, Gateway, as a microservice, also exposes this in different ways depending on the protocol used.
+
+
+### Gateway health check with grpc
+
+When using grpc as the protocol to communicate with the Gateway, then Gateway uses the exact same mechanism as Executors to expose their individual health status. It exposes [grpc health check](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) to the outside world.
+
+With the same Flow described before we can use the same way to check the gateway status:
+
+```bash
+docker run --network='host' fullstorydev/grpcurl -plaintext 127.0.0.1:12345 grpc.health.v1.Health/Check
+```
+
+```text
+{
+  "status": "SERVING"
+}
+```
+
+
+### Gateway health check with http or websocket
+
+When using grpc or http as a procotol for the Gateway, then it exposes and endpoint '/' that one can query to check the status.
+
+By having a Flow with http or websocket as protocol:
+
+```python
+from jina import Flow
+
+f = Flow(protocol='http', port=12345).add()
+with f:
+    f.block()
+```
+
+```bash
+curl http://localhost:12345
+```
+
+Then you will get a valid empty response indicating its ability to serve.
+```text
+{}%
+```
+
+## Readiness of a Flow exposed to the client
+
+A lot of times, from the client perspective, it is useful to know if a Flow, as a complete set of microservices, is ready to receive requests. This is why the gateway 
+exposes an endpoint for each of the supported protocols to know the health and readiness of a Flow. 
+
+Also Jina Flow and Client contains a convenient API to query these endpoints. You can call `flow.dry_run()` and `client.dry_run()`
```

Comment:
```suggestion
Jina `Flow` and `Client` offer a convenient API to query these readiness endpoints. You can call `flow.dry_run()` and `client.dry_run()`, which will return `True` if the Flow is healthy and ready, and `False` otherwise.
```